### DISCUSSION TITLE
Remember that tslab users may be new to Jupyter

### DISCUSSION THREAD

**Discussion by TonyGravagno** [yunabe/tslab#96] [Ideas]
I'm a JS/TS developer, familiar but not intimate with Python, and completely new to Jupyter. I have a few challenges here that I'm working through and I hope these notes help with ongoing development.

This has been my on-ramp to this platform, and I believe others would have a similar experience:

- See Jupyter being used in videos and articles.
- Decide to research.
- Visit jupyter.org.
- Install JupyterLab or Jupyter Notebook, maybe with a bit of struggle with Python.
- Learn what a kernel is, in this context.
- Look at the [wiki Kernel list](https://github.com/jupyter/jupyter/wiki/Jupyter-kernels) to find something for JS/TS.
- Find tslab.
- **Profit!** :grin:

So I installed and verified Jupyter, but not being a skilled Python devloper I can't actually use it until I go through the tslab README to install this package.

The README and other notes here are written by Jupyter users, for Jupyter users. It is reasonable to assume that many JS/TS devs who come here are not familiar with Python, and may have never used Jupyter before this. So **tslab** may be their very first exposure to the Jupyter ecosystem. I think it's important for adoption of this kernel as well as for Jupyter that the developers here recognize this situation and incorporate this understanding into project notes, README, wiki, etc.

"We're not here to teach Jupyter"

That's true, but by definition, someone who wants to use tslab is a JS/TS developer (or wants to be) and therefore we may not know anything about Python. We can't use Jupyter until we have a gateway tool like tslab. These things go hand-in-hand.

I'm not asking for documentation or tutorials or anything else that duplicates what's already available in the platform ecosystem. I'm just saying tslab and all other kernel extensions play an important role in introducing the platform. Many newcomers **must** come through a kernel like this to begin using the platform. So I think it's important for all kernel providers to recognize their position in the ecosystem.

Here is another example of why tslab developers/users need to recognize these relationships:

When I open a new Notebook, the first thing I typed was `console.log("hi");`. What a downer. ☹️ 

> Cannot find name 'console'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.

"target"? Hmm, this needs a package.json, maybe a tsconfig.json. That makes sense, but there's nothing in the README or elsewhere about this. I used `npm init` and `npm install typescript`, verified that tsconfig.json includes "dom", and the error persists.

So I have no idea what the problem is yet. I'll continue to research but I hope someone provides the magic tip here. 😁 Without more info beyond the README, I have no idea what other issues can be expected. With no response to #90 I don't even know if I'm trying a supported platform.

Bottom line : Please remember how people get here and what understandings/expectations are when people are trying to use this software for the first time. New versions certainly work in your own development systems with your own established configurations, but none of the rest of us have this pre-installed, so we're depending on you to provide the info required.

Thanks!

**Comment by iSenses** [comment]
> 
> Here is another example of why tslab developers/users need to recognize these relationships:
> 
> When I open a new Notebook, the first thing I typed was `console.log("hi");`. What a downer. ☹️
> 
> > Cannot find name 'console'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.
> 
Met the same problem, will you please share how you solve it?
